packages/store/src/Hook.sol

```diff
@@ -14,6 +16,36 @@ library HookLib {
     // Move the address to the leftmost 20 bytes and the bitmap to the rightmost byte
     return Hook.wrap(bytes21(bytes20(hookAddress)) | bytes21(uint168(encodedHooks)));
   }
+
+  /**
+   * Filter the given hook from the hook list at the given key in the given hook table
+   */
+  function filterListByAddress(bytes32 hookTableId, bytes32 key, address hookAddressToRemove) internal {
+    bytes21[] memory currentHooks = Hooks.get(hookTableId, key);
```

Comment:
Is it better/more gas efficient to do this than to pass in the array and get an array back out?

Reply:
yeah, unfortunately much more gas efficient to pass as little memory around as possible - passing arrays is pretty expensive. (Otherwise would have preferred a more functional approach with arrays as input/output too)